### PR TITLE
triggerhappy: update to upstream version 0.5.0

### PR DESCRIPTION
**Pull request by wertarbyte**
Signed-off-by: Stefan Tomanek <stefan.tomanek@wertarbyte.de>
-------------------------------

Maintainer: me / @wertarbyte
Compile tested: ar71xx, TP-Link Archer C7, LEDE 17.01.2
Run tested: ar71xx, TP-Link Archer C7, LEDE 17.01.2, tested basic functionality
Description:  Update to latest upstream version and fix licence information

### CODE REVIEW

File: utils/triggerhappy/Makefile [hunk 1]
```diff
@@ -8,18 +8,17 @@
...
 PKG_NAME:=triggerhappy
-PKG_VERSION:=0.3.4-151001
-PKG_REV:=7e5abc69f215678e93a6b999524981c8b40bdcd9
+PKG_VERSION:=0.5.0
+PKG_REV:=b822888066129350e51ad79f1cf307fa38dae4f7
 PKG_RELEASE:=1
 
-PKG_SOURCE:=$(PKG_NAME)-$(PKG_VERSION).tar.gz
-PKG_SOURCE_URL:=git://github.com/wertarbyte/triggerhappy
+PKG_SOURCE_URL:=https://github.com/wertarbyte/triggerhappy
```
Comment (diizzyy) [inline utils/triggerhappy/Makefile]: Please add PKG_MIRROR_HASH

Comment (wertarbyte) [inline utils/triggerhappy/Makefile]: Done

**Comment by diizzyy** [comment]
..and squash commits ;-)

**Comment by thess** [comment]
Since this is a released package located at:
https://github.com/wertarbyte/triggerhappy/archive/release/0.5.0.tar.gz

 I'd like to see a fetch of the tar.gz and not do a direct git checkout when it can be avoided. PKG_REV is un-necessary then.

**Comment by diizzyy** [comment]
That package is also clearly not usable as 0.5.0.tar.gz is going to clash with <insert something here>. The distfiles solution is fine and fixes shortcomings with the build system.

**Comment by wertarbyte** [comment]
I have to find out how to keep github from creating tarballs ("releases") for every git tag I push to the repository :)

File: utils/triggerhappy/Makefile (hunk 1 above)
Comment (wertarbyte) [inline utils/triggerhappy/Makefile]: (and squashed the commits)

**Comment by thess** [comment]
@wertarbyte, @diizzyy -- For future reference, there is nothing preventing one from adding another package or any attachment to a github release. I do this often myself. Just edit the _release_ item and drop it there.

Ex: https://github.com/thess/pianod-sc/releases

Notice the nice .xz file  ;)

**Comment by stintel** [comment]
Just use https://github.com/wertarbyte/triggerhappy/archive/release/0.5.0/triggerhappy-0.5.0.tar.gz (you can name the last part however you want)

**Comment by wertarbyte** [comment]
OK, I modified the Makefile to use the HTTP(s) download from Github.

**Comment by jow-** [comment]
Merged, thanks.

**Pull request merged**